sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -398,7 +398,7 @@ hintStatement
     ;
 
 fromClause
-    : FROM relation (',' relation)* (pivotClause | lateralView*)?
+    : FROM relation (',' relation)* lateralView* pivotClause?
```

Comment:
as I said in the e-mail in the dev list, I think that the right fix is `(pivotClause | lateralView+)?`. I have been running the tests and they are successful so far, so I think this is the right approach. It also avoids the additional check needed to throw the ParseException.

Reply:
Yes, I agree that "(pivotClause | lateralView+)?" would also be a fix for the problem. But by doing a check in a later stage would give a more detailed and meaningful exception.

Reply:
I think the more constraints we can express in the parser, the better. @gatorsmile @maropu what do you think?

Reply:
FYI, I am doing this based on the ORACLE syntax definition. This "no PIVOT and LATERAL together" thing is listed as a restriction rather than reinforced in the parser rules in ORACLE. That said, I believe I should have done this in my first check-in.

Reply:
@mgaido91 Better readability is more important to our end users. Currently, we rely on ANTLR4 to issue the error message in most cases. However,  ANTLR4 is not good to issue a readable error message, compared with the commercial vendor. 

We need to revisit what our current Parser implementation and improve the error handling.